internal/lock.go

```diff
@@ -26,178 +28,295 @@ type config struct {
 	Resource []Resource
 }
 
+// NewLock creates a new lock file instance
 func NewLock(path string, newOk bool) (*Lock, error) {
 	_, error := os.Stat(path)
 	if os.IsNotExist(error) {
 		if newOk {
-			return &Lock{path: path}, nil
+			return &Lock{path: path, conf: config{Resource: []Resource{}}}, nil
 		} else {
 			return nil, fmt.Errorf("file '%s' does not exist", path)
 		}
 	}
+
 	var conf config
 	file, err := os.Open(path)
 	if err != nil {
-		return nil, error
+		return nil, err
 	}
+	defer file.Close()
+
 	d := toml.NewDecoder(file)
-	err = d.Decode(&conf)
-	if err != nil {
-		return nil, err
+	if err := d.Decode(&conf); err != nil {
+		return nil, fmt.Errorf("failed to decode lock file: %w", err)
```

Comment:
I'll say this one more time on this PR and then stop making the same comment repeatedly: this is a style change that doesn't really have anything to do with the Artifactory functionality.  If you want to propose this change, put it in a separate PR.  Go ahead and remove this change from this PR.